packages/amplify-category-hosting/index.js

```diff
@@ -61,8 +61,11 @@ function publish(context, service, args) {
   const { enabledServices } = categoryManager.getCategoryStatus(context);
   if (enabledServices.length > 0) {
     if (enabledServices.includes(service)) {
-      if (service === 'S3AndCloudFront' && context.exeInfo?.amplifyMeta.hosting.S3AndCloudFront.output) {
-        if (context.exeInfo.amplifyMeta.hosting.S3AndCloudFront.output['CloudFrontSecureURL'] === undefined) {
+      // checks if hosting with S3 and CloudFront and has a secure URL otherwise,
+      // the user is hosting from S3 without CloudFront distribution and should throw an error
+      if (service === 'S3AndCloudFront') {
+        let cfnOutput = context.exeInfo?.amplifyMeta.hosting.S3AndCloudFront.output;
```

Comment:
nit const. And should this be an optional chain?

Reply:
should be optional because when first publish occurs, the context does not have any amplifyMeta information. This was causing unit tests to fail. Another option could be to modify the tests to account for that. Not exactly sure how I would go about that.